pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -790,7 +790,10 @@ public CompletableFuture<Void> seekAsync(MessageId messageId) {
     public CompletableFuture<Void> seekAsync(long timestamp) {
         List<CompletableFuture<Void>> futures = new ArrayList<>(consumers.size());
         consumers.values().forEach(consumer -> futures.add(consumer.seekAsync(timestamp)));
-        return FutureUtil.waitForAll(futures);
+        return FutureUtil.waitForAll(futures).thenApply(f -> {
+                    clearIncomingMessages();
```

Comment:
Is there a scenario like this?
- inner-consumer-1 seeks finished
- inner-consumer-2 seeks finished
- receive messages from inner-consumer-1, and push them to incoming-queue
- clear incoming-queue

then some messages are lost.

Reply:
I'll check this.

Reply:
You're right, there is a chance for data loss.
Change to draft first.

Reply:
Update, PTAL again.

Reply:
If a task runs at this line below, we call it `receiving-task`.

https://github.com/apache/pulsar/blob/f9af4245e0b05c382656fc674fdaeda26487258c/pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java#L244

Then is there a scenario like this?

| time | `receiving-task` | `seek task` |
| --- | --- | --- |
| 1 | `receiving-task` started, and it is not finished | | 
| 2 |  | seek | 
| 3 |  | clear the `incoming queue` after seeking |
| 4 | `receiving-task` finished, and push messages to `incoming queue` |

Then the consumer receives duplicated messages